docs/designs/topology.md

```diff
@@ -563,11 +645,21 @@ The operator translates user's level names to keys and builds required/preferred
 **Preferred Constraints (Auto-Generated):**
 
 - Operator ALWAYS generates preferred constraint at all three levels
-- Uses key of strictest level (e.g., `"kubernetes.io/hostname"` for "host" level)
+- Uses key of strictest (narrowest) level configured in ClusterTopology
+- Example: If levels include "host", preferred = `"kubernetes.io/hostname"`
 - Writes to PodGang: `TopologyConstraint.PackConstraint.Preferred = "kubernetes.io/hostname"`
 - Enables out-of-box optimization even without user configuration
 - Scheduler can fallback to less strict levels if preferred cannot be satisfied
 
+**Edge Cases:**
+
+1 **Strictest Level Changes During Topology Update**:
+   - Initial topology: rack, host (strictest = host)
+   - Updated topology: rack, host, numa (strictest = numa)
```

Comment:
Hmm, just realized I don't completely understand the numa domain. Maybe we should just define all the domains and their hierarchy/order near the top for reference?